addon/ng2/utilities/module-resolver.ts

```diff
@@ -0,0 +1,113 @@
+   * 
+   * @todo Implement the logic for rewriting imports of the dependent files when the file
+   *       being moved has index file in its old path and/or in its new path.
+   * 
+   * @return {Promise<Change[]>} 
+   */
+  resolveDependentFiles(): Promise<Change[]> {
+    return dependentFilesUtils.getDependentFiles(this.oldFilePath, ROOT_PATH)
+      .then((files:  dependentFilesUtils.ModuleMap) => {
+        let changes: Change[] = [];
+        Object.keys(files).forEach(file => {
+          let tempChanges: ReplaceChange[] = files[file]
+            .map(specifier => {
+              let fileDir = path.dirname(file);
+              let changeText = path.relative(fileDir, this.newFilePath);
+              if (changeText.length > 0 && changeText.charAt(0) !== '.') {
+                changeText = `.${path.sep}${changeText}`;
+              }
+              if (changeText === '') {
+                changeText = `.${path.sep}${path.basename(this.oldFilePath, '.ts')}`;
+              } else {
+                changeText = `${changeText}${path.sep}${path.basename(this.oldFilePath, '.ts')}`;
+              }
+              let position = specifier.end - specifier.specifierText.length;
+              return new ReplaceChange(file, position - 1, specifier.specifierText, changeText);
+            });
+          if (tempChanges.length > 1) {
+            tempChanges = tempChanges.sort((currentChange, nextChange) => {
+              return nextChange.order - currentChange.order;
+            });
+          };
+          changes = changes.concat(tempChanges);
+        });
+        return changes;
+      });
+ }
+
+  /**
+   * Rewrites the file's own relative imports after it has been moved to new Path.
+   * 
+   * @return {Promise<Change[]>}
+   */
+  resolveOwnImports(): Promise<Change[]> {
+    return dependentFilesUtils.createTsSourceFile(this.oldFilePath)
+      .then((tsFile: ts.SourceFile) => dependentFilesUtils.getImportClauses(tsFile))
+      .then(moduleSpecifiers => {
+        let changes: Change[] = moduleSpecifiers
+        .filter(importClause => {
+            // Filter only relative imports.
+            let singleSlash = importClause.specifierText.charAt(0) === '/';
+            let currentDirSyntax = importClause.specifierText.slice(0, 2) === './';
+            let parentDirSyntax = importClause.specifierText.slice(0, 3) === '../';
+            return singleSlash || currentDirSyntax || parentDirSyntax;
+          })
+          .map(specifier => {
+            let specifierText = specifier.specifierText;
+            let moduleAbsolutePath = path.resolve(path.dirname(this.oldFilePath), specifierText);
+            let changeText = path.relative(this.newFilePath, moduleAbsolutePath);
+            if (changeText.length > 0 && changeText.charAt(0) !== '.') {
+              changeText = `.${path.sep}${changeText}`;
+            }
+            changeText = (changeText === '') ? `.${path.sep}${path.basename(this.oldFilePath)}` : changeText;
```

Comment:
I found a way to never get a empty `changeText`, so I have removed this if condition.